packages/client/lib/miner/miner.ts

```diff
@@ -0,0 +1,258 @@
+      // will not have been initialized yet in the execution vm
+      // since the following line won't be reached:
+      // https://github.com/ethereumjs/ethereumjs-monorepo/blob/c008e8eb76f520df83eb47c769e3a006bc24124f/packages/client/lib/sync/execution/vmexecution.ts#L100
+      // So we will do it here:
+      await vmCopy.stateManager.generateCanonicalGenesis()
+    } else {
+      // Set the state root to ensure the resulting state
+      // is based on the parent block's state
+      await vmCopy.stateManager.setStateRoot(parentBlockHeader.stateRoot)
+    }
+
+    let difficulty
+    if (this.config.chainCommon.consensusType() === ConsensusType.ProofOfAuthority) {
+      // Determine if signer is INTURN (2) or NOTURN (1)
+      const inTurn = await vmCopy.blockchain.cliqueSignerInTurn(signerAddress)
+      difficulty = inTurn ? 2 : 1
+    }
+
+    let baseFeePerGas
+    const londonHardforkBlock = this.config.chainCommon.hardforkBlockBN('london')
+    const isInitialEIP1559Block = londonHardforkBlock && number.eq(londonHardforkBlock)
+    if (isInitialEIP1559Block) {
+      baseFeePerGas = new BN(this.config.chainCommon.param('gasConfig', 'initialBaseFee'))
+    } else if (this.config.chainCommon.isActivatedEIP(1559)) {
+      baseFeePerGas = parentBlockHeader.calcNextBaseFee()
+    }
+
+    const parentBlock = (this.synchronizer as any).chain.blocks.latest
+    const blockBuilder = await vmCopy.buildBlock({
+      parentBlock,
+      headerData: {
+        number,
+        difficulty,
+        gasLimit,
+        baseFeePerGas,
+      },
+      blockOpts: {
+        cliqueSigner: signerPrivKey,
+        hardforkByBlockNumber: true,
+      },
+    })
+
+    const txs = await this.synchronizer.txPool.txsByPriceAndNonce(
+      vmCopy.stateManager,
+      baseFeePerGas
+    )
+    this.config.logger.info(
+      `Miner: Assembling block from ${txs.length} eligible txs ${
+        baseFeePerGas ? `(baseFee: ${baseFeePerGas.toNumber()})` : ''
+      }`
+    )
+    let index = 0
+    let blockFull = false
+    while (index < txs.length && !blockFull && !interrupt) {
+      try {
+        await blockBuilder.addTransaction(txs[index])
+      } catch (error: any) {
+        if (error.message === 'tx has a higher gas limit than the remaining gas in the block') {
+          if (blockBuilder.gasUsed.gt(gasLimit.subn(21000))) {
+            // If block has less than 21000 gas remaining, consider it full
+            blockFull = true
+            this.config.logger.debug(
+              `Miner: Assembled block full (gasLeft: ${gasLimit
+                .sub(blockBuilder.gasUsed)
+                .toNumber()})`
+            )
+          }
+        } else {
+          // If there is an error adding a tx, it will be skipped
+          const hash = '0x' + txs[index].hash().toString('hex')
+          this.config.logger.error(
+            `Skipping tx ${hash}, error encountered when trying to add tx:\n${error}`
+          )
```

Comment:
Also no blocker, but are these likely errors in our code (then this would be ok) or errors with malformed txs submitted to the tx pool (then we should rather do `logger.debug()`?

Reply:
good point i'll update to `debug`, yes these would be errors from malformed txs in the pool, ideally we would regularly check txs in the pool and validate/invalidate based on the current state as to not take up precious time during block assembly, however i figure we don't have to worry about mainnet load or speed quite yet and can optimize later as needed :)